### PR TITLE
fix(systemd): check if service is enabled before restarting

### PR DESCRIPTION
**Pull request by StealthyCoder**
A service will have Substate be equal to `dead` and ActiveState be equal to `inactive` if it is **disabled**. If it is _enabled_ then it will be Substate `exited` for example but ActiveState will be `active`.

### CODE REVIEW

File: internal/rotate.go [hunk 1]
```diff
@@ -196,19 +196,26 @@ func (h *CertRotationHandler) RestartServices() {
...
+	services := []string{"aktualizr-lite.service", "fioconfig.service"}
+	units, err := con.ListUnitsByNamesContext(ctx, services)
+	if err != nil {
+		log.Fatalf("Unable to query status of units: %s", err)
+	}
+	for _, unit := range units {
 		restartChan := make(chan string)
-		_, err = con.RestartUnitContext(ctx, svc, "replace", restartChan)
+		if unit.ActiveState == "inactive" || unit.SubState == "dead" {
+			continue
```
Comment (doanac) [inline internal/rotate.go:207]: I think i'd rather restart it anyway. If this is the fioconfig daemon it needs to restart or it won't be able to talk to the server anymore.

Comment (StealthyCoder) [reply internal/rotate.go:207]: So the discussion was indeed actually only concerning `aklite` so I will amend that to be the case.

File: internal/rotate.go [hunk 2]
```diff
@@ -196,9 +196,17 @@ func (h *CertRotationHandler) RestartServices() {
 	if err != nil {
 		log.Fatalf("Unable to connect to DBUS for service restarts: %s", err)
 	}
-
-	for _, svc := range []string{"aktualizr-lite.service", "fioconfig.service"} {
+	services := []string{"aktualizr-lite.service", "fioconfig.service"}
```
Comment (mike-sul) [inline internal/rotate.go:199]: I think we don't need to check the `fioconfig` service/unit status, just the aklite's one needs to be checked.

Comment (StealthyCoder) [reply internal/rotate.go:199]: Already changed :+1:

Comment (doanac) [reply internal/rotate.go:199]: If we are going down this rabbit hole. I assume this about people not running the daemon. If you run a custom sota client - I think we need logic for that. We probably need to do #define type thing so this value can be overridden.

It's a bigger piece of work and requires some logic in the yocto recipe. Not sure if we rush to fix this now or wrap up scale test work and then do this. v93 is basically done - so we have a long time to get this fixed.

Comment (mike-sul) [reply internal/rotate.go:199]: I am not sure this is a rabbit hole. IMO, this is straightforward - just don't restart aklite or custom sota client daemons if their services are not defined or disabled. There are customers that don't set an update agent in a daemon mode or "hold" via the callbacks.

Comment (doanac) [reply internal/rotate.go:199]: k. merge this as is and come back and add support to handle a custom sota client. then update yocto to build this with the correct value. My bigger point was that I think Eric can finish his other task first and then come back to this.

Comment (mike-sul) [reply internal/rotate.go:199]: I agree on the priority for sure.

**Comment by doanac** [comment]
k - this is pretty late in the release cycle and we have a different PR that's a must fix - https://github.com/foundriesio/fioconfig/pull/50

I'm not sure we want to ship this the next release do we?

**Pull request merged**